modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/files_info_query.rb

```diff
@@ -132,13 +132,15 @@ def create_storage_file_infos
 
     # rubocop:enable Metrics/AbcSize
 
-    def location(file_path)
+    def location(file_path, mimetype)
       prefix = 'files/'
       idx = file_path.rindex(prefix)
       return '/' if idx == nil
 
       idx += prefix.length - 1
-
+      # Remove the following when /filesinfo starts responding with a trailing slash for directory paths
+      # in all supported versions of OpenProjectIntegation Nextcloud App.
+      file_path << '/' if mimetype == 'application/x-op-directory' && file_path[-1] != '/'
```

Comment:
🟡 why does this fix doesn't require amending the nextcloud files info query spec??

-> Answer: because it currently doesn't assert the real data. But it should. Please compare to the one drive files info query spec.

Reply:
The request are stubbed, but they are stubbed with actual responses there. I added appropriate location expectation to the spec.